### PR TITLE
connected project to DB realised ProductDaoJdbcImpl

### PR DESCRIPTION
**Pull request by Oleh888**

### CODE REVIEW

File: src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Override
+    public List<Product> getAll() {
+        List<Product> products = new ArrayList<>();
+        String sqlGetAll = "SELECT * FROM products";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(sqlGetAll);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Product product = new Product(resultSet.getString("product_name"),
```
Comment (boroda4436) [inline src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java:67]: code duplication on line 48

File: src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Product product = new Product(resultSet.getString("product_name"),
+                            resultSet.getDouble("price"));
+                product.setId(id);
+                return Optional.of(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't SELECT product in DB with id: " + id, e);
```
Comment (boroda4436) [inline src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java:54]:
```suggestion
            throw new DataProcessingException("Can't SELECT product from DB with id: " + id, e);
```

File: src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.Optional;
+import mate.academy.dao.ProductDao;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Product;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    private static final int ID_COLUMN = 1;
```
Comment (boroda4436) [inline src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java:19]: nope. Use column name, not column position

File: src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (Oleh888) [reply src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java:67]: created appropriate method for this

File: src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java (hunk 3 above)
Comment (Oleh888) [reply src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java:19]: done

**Pull request merged**